aws-s3-transfer-manager/src/operation/download.rs

```diff
@@ -115,13 +115,37 @@ async fn send_discovery(
     }
 
     // acquire a permit for discovery
-    let permit = ctx.handle.scheduler.acquire_permit().await?;
+    let permit = ctx.handle.scheduler.acquire_permit().await;
+    let permit = match permit {
+        Ok(permit) => permit,
+        Err(err) => {
+            if comp_tx.send(Err(err)).await.is_err() {
+                tracing::debug!("Download handle for key({}) has been dropped, aborting during the discovery phase", input.key.unwrap_or("None".to_string()));
```

Comment:
**fix**: You should be able to just pass `input.key` if you change the format to `key({:?})`

e.g.

```rs
tracing::debug!("Download handle for key({:?}) has been dropped, aborting during the discovery phase", input.key);
```

Same thing elsewhere

Reply:
Thanks, updated.